src/drivers/dshot/DShot.h

```diff
@@ -0,0 +1,234 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2019-2020 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+#pragma once
+
+#include <drivers/device/device.h>
+#include <drivers/drv_input_capture.h>
+#include <drivers/drv_mixer.h>
+#include <lib/mixer_module/mixer_module.hpp>
+#include <px4_platform_common/getopt.h>
+#include <px4_platform_common/module.h>
+#include <uORB/topics/esc_status.h>
+
+#include "DShotTelemetry.h"
+
+using namespace time_literals;
+
+#if !defined(BOARD_HAS_PWM)
+#  error "board_config.h needs to define BOARD_HAS_PWM"
+#endif
+
+/** Dshot PWM frequency, Hz */
+static constexpr unsigned int DSHOT150  =  150000u;
+static constexpr unsigned int DSHOT300  =  300000u;
+static constexpr unsigned int DSHOT600  =  600000u;
+static constexpr unsigned int DSHOT1200 = 1200000u;
+
+static constexpr int DSHOT_DISARM_CMD   = dshot_command_t::DShot_cmd_motor_stop;
```

Comment:
This is confusing, it's not the same thing. Command != throttle value. The disarmed value is a throttle, and if the output requests that, it get's changed to a stop command (see `updateOutputs`).
So I'd rather keep the previous implementation, which reflected that correctly (with the +1, which might still be a separate define).
```
static constexpr int DSHOT_DISARMED_THROTTLE   = 0;
static constexpr int DSHOT_MIN_THROTTLE = DSHOT_DISARMED_THROTTLE + 1;
static constexpr int DSHOT_MAX_THROTTLE = 1999;
```

Reply:
Cool. I can do that!

Reply:
However, given the dshot protocol, 0 is a command, not a valid throttle... so I'd like to keep the name as CMD.